### ISSUE TITLE
JSDoc "nameless" @typedef on declaration does not work inside its own module

### ISSUE THREAD

**Issue by ikabirov** [microsoft/TypeScript#26626]
index.js
```
import {MyType} from "./MyType";

/**
 * @param {MyType} article
 */
function test(article) {
	article.custom; // error TS2339: Property 'custom' does not exist on type '{ src: string; }'.
}
```

MyType.js
```
/**
 * @typedef {{src: string}}
 */
let MyType;

/**
 * @param {MyType} article
 */
function test(article) {
	article.custom; // unknown type, no error
}

export {
	MyType,
}
```

**Comment by KeithHenry** [comment]
Shouldn't it be (in `MyType.js`):
``` js
/**
 * @typedef {Object} MyType
 * @property {string} src
 */

/**
 * @param {MyType} article
 */
function test(article) {
	article.custom; // unknown type, no error
}

export {
	MyType,
}
```

**Comment by ikabirov** [comment]
No, it does not supported by Closure Compiler

**Comment by andy-ms** [comment]
Hm...
```ts
/** @typedef {{src: string}} */
let T;

/** @type {T} */
let x = JSON.parse("whatever");
x.brc; // No error?

export { T };
```

`x` is `any` despite having a type annotation -- but it works if the `export` is removed.

**Comment by ikabirov** [comment]
seems without export TS process file.js like standalone file, not ES6 module.

**Comment by jameskeane** [comment]
just a heads up, @sandersn I've already started working on this, I'll have a patch tomorrow probably.

**Comment by sandersn** [comment]
This is fixed in 3.8.
